app/Http/Controllers/BookTitleController.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\BookTitle;
+use App\Models\Genre;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+
+class BookTitleController extends Controller
+{
+    public function index(Request $request)
+    {
+        $filtered = BookTitle::query();
+
+        if ($request->has('name')) {
+            $filtered->whereRaw("LOWER(name) LIKE '%" . strtolower($request->name) . "%'");
```

Comment:
Không nên ghép chuỗi để tránh các lỗi SQL Injection.
```suggestion
            $filtered->where(DB::raw('LOWER(name)'), 'LIKE', '%' . strtolower($request->name) . '%')->toSql();
```